app/models/topic_tracking_state.rb

```diff
@@ -480,8 +480,7 @@ def self.report_raw_sql(
       LEFT JOIN category_users ON category_users.category_id = topics.category_id AND category_users.user_id = #{user.id}
       LEFT JOIN dismissed_topic_users ON dismissed_topic_users.topic_id = topics.id AND dismissed_topic_users.user_id = #{user.id}
       #{additional_join_sql}
-      WHERE u.id = :user_id AND
```

Comment:
If we look at the joins, we are already joining with `JOIN users u on u.id = :user_id`.

Reply:
there are potential performance impacts here, a bit concerned about fiddling with this, can you confirm query plan is identical for a user with lots of state?

Reply:
I tested this locally and both query plans are the same.

Reply:
Can you test on a few people with large amounts of read state on say meta?

Reply:
I think that for inner joins, multiple FROM, JOIN … ON or JOIN … WHERE are the same. I think Postgres will rewrite this query into lists of FROM anywhere because of:

https://www.postgresql.org/docs/current/runtime-config-query.html#GUC-JOIN-COLLAPSE-LIMIT